dgraph/cmd/zero/raft.go

```diff
@@ -458,7 +458,7 @@ func (n *node) proposeNewCID() {
 	}
 
 	// Apply trial license only if not already licensed.
-	if n.server.license() == nil {
+	if n.server.license() == nil && Zero.Conf.GetString("enterprise_license") == "" {
```

Comment:
The license will be applied for the second time in node.Run()
https://github.com/dgraph-io/dgraph/blob/1783e0edafb995b344eb5cfd22008f9ab0b29be8/dgraph/cmd/zero/raft.go#L795-L803
Is it okay to reapply the license? I think the other call would complain.

I think we apply the trial license here so that we can get a CID but I'm not 100% sure.

Reply:
The problem is trail license is applied after enterprise license in some cases. I thought the same as moving the trail license to Run method. But, this is little cleaner and simple approch.

Reply:
How about moving the `apply trial` code from here to where we actually apply the license?
So we apply either the license user has provided or the trial license.

Reply:
That would lead to applying the trail license again and again. I believe they kept the trail code in CID generation because they want to apply the trail only when the CID is generated.

Reply:
Okay. This should be okay to do in that case.